File: find.java
//Sanakirjatiedostolistat english ja finnish 
//ja niiden vertailu, samojen sanojen etsiminen
import java.util.*;
import java.io.*;

public class Find {
    public static void main(String[] args) throws Exception {
        ArrayList<String> englishWords = new ArrayList<>();
        HashSet<String> finnishWords = new HashSet<>();
        Scanner reader = new Scanner(new File("english.txt"));
        while (reader.hashNextLine()) {
            englishWords.add(reader.nextLine());
        }
        reader = new Scanner(new File("finnish.txt"));
        while (reader.hashNextLine()) {
            finnishWords.add(reader.nextLine());
        }
        int counter = 0;
        long startTime = System.nanoTime();
        for (String word : englishWords) {
            if (finnishWords.contains(word)) {
                System.out.println(word);
                counter++;
            }
        }
        long endTime = System.nanoTime();
        System.out.println("Sanoja yhteensä: " + counter);
        System.out.println("Aikaa hakuun kului: " + (endTime-startTime)/1e9 + " s.");
    }
}

File: find.py
//Sanakirjatiedostolistat english ja finnish 
//ja niiden vertailu smojen sanojen etsiminen
from time import time

english_words = open("english.txt").readlines()
finnish_words = set(open("finnish.txt").readlines())

counter = 0
startTime = time()
for word in english_words:
    if word in finnish_words:
        print(word.strip())
        counter += 1
endTime = time()

print("Sanoja yhteensä:", counter)
print("Aikaa kului:", endTime-startTime/1e9, "s.")
